src/Arcus.Security.Core/Providers/EnvironmentVariableSecretProvider.cs

```diff
@@ -55,7 +59,10 @@ public Task<Secret> GetSecretAsync(string secretName)
         /// <exception cref="T:Arcus.Security.Core.SecretNotFoundException">The secret was not found, using the given name</exception>
         public Task<string> GetRawSecretAsync(string secretName)
         {
-            Guard.NotNullOrWhitespace(secretName, nameof(secretName), "Requires a non-blank secret name to look up the environment secret");
+            if (string.IsNullOrWhiteSpace(secretName))
+            {
+                throw new ArgumentNullException(nameof(secretName), "Requires a non-blank secret name to look up the environment secret");
```

Comment:
I believe we should throw an ArgumentException instead of an ArgumentNullException if the test for `String.IsNullOrWhiteSpace` evaluates to true, just to be consistent accross the code-base ?

I believe `Guard.NotNullOrWhitespace` is a bit more sophisticated as it looks like it throws an `ArgumentNullException` when the argument is null, and otherwise it throws an `ArgumentException`.

Reply:
Indeed! I have missed these files when adapting the exceptions. I've corrected them now

Reply:
Thanks!